install_files/ansible-base/roles/tails-config/tasks/migrate_existing_config.yml

```diff
@@ -0,0 +1,33 @@
+---
+# Between 0.3.12 and 0.4, the "Document Interface" was renamed to
+# "Journalist Interface". Since multiple scripts expect to readfrom this file,
+# we'll rename the old file to handle the transition gracefully.
+- name: Migrate Document Interface ATHS file (upgrade only).
+  command: >-
+    mv
+    "{{ tails_config_ansible_base }}/app-document-aths"
+    "{{ tails_config_ansible_base }}/app-journalist-aths"
+  args:
+    # Only run this command if the deprecated file is present.
+    removes: "{{ tails_config_ansible_base }}/app-document-aths"
+
+# Between 0.3.12 and 0.4, the `prod-specific.yml` vars file was deprecated
+# in favor of `group_vars/all/site-specific`. The latter is not version
+# controlled, and so will not exist by default.
+- name: Migrate site-specific configuration to new location.
+  command: >-
+    cp
```

Comment:
Oh, there's a very definite reason we're using `cp` and not `mv` here: the `prod-specific.yml` file is version-controlled, whereas the newer `group_vars/all/site-specific` file is not. Version controlling the template for the prod-specific.yml file has caused a lot of headaches around iterating on the configuration, since Admins must stash changes, pull the latest tag, then pop their changes, and pray there aren't any conflicts during pop.

So a `mv` would break the version control on that file (a `git mv` would be the right approach), and cause further problems down the road with merge conflicts. It may be worthwhile instructing Admins to `git checkout prod-specific.yml` to completely wipe the customizations on that file. That I'd consider, but I didn't request it because we can't be certain Admins have dutifully filled out their KeePass database file to document all the secrets required for instance access—thereby giving them a simple recovery path in the event information is lost from the prod-specific.yml file.

Since we're copying the file, however, I suppose advising a `git checkout prod-specific.yml` file is indeed appropriate. What say you, @msheiny?